### DISCUSSION TITLE
Usage as a place for enhanced documentation.

### DISCUSSION THREAD

**Discussion by ioquatix** [community/community#4703] [Discussions] [answered]
Hello,

I want to use GitHub Discussions for extended documentation.

The way this would work, is we have static documentation which is generated from the source code. We would then link to GitHub discussion, e.g.

```
https://github.com/socketry/async/discussions/for/Async::Reactor#run
```

This would result in the following:

- If there is a discussion with that exact title, redirect to it.
- If there is no discussion with that title, prompt the user to create one.

To extend this idea further, the ability to **embed** a discussion in the same way, e.g.

```
https://github.com/socketry/async/discussions/embed/Async::Reactor#run
```

This would either embed any discussion matching the given title, or present a form by which the user can start it.

Other ways to do this would involve pre-set title, category, etc.

This would be a massive interoperability win IMHO, since it would allow GitHub static pages to provide meaningful dynamic content.

**Comment by laymonage** [comment, marked as answer]
Hey @ioquatix, I built a project that seems to accomplish exactly what you want: check out https://giscus.app 😃 

You can see an example of a website that utilizes my app to enhance the book content with discussions: https://os.phil-opp.com/freestanding-rust-binary/

**Reply by ioquatix** [reply]
Oh, that's pretty sweet. Can I embed this into GitHub Pages directly or do I need to use an external GitHub App?

**Reply by laymonage** [reply]
@ioquatix You'll need to install the app on your repository (can be a separate repo), then you can just drop the generated `<script>` tag to your webpage. Users will need to authenticate with the app in order to leave a comment, though.

**Reply by ioquatix** [reply]
I see, that makes sense. What you created here is really awesome, it's basically exactly what I wanted, and I have to say the whole setup process is very slick and easy to understand.

I would say that in terms of the integration, it's pretty good. It would be nice if signing in didn't need a page reload.

I've integrated it directly into my documentation generation tool for Ruby: https://github.com/socketry/utopia-project

It's now up and running here: https://socketry.github.io/async/source/Async/Clock/index.html (scroll to the bottom).

I'm not sure what UX would work best, but I'm fine with this to start with. Let me know if you have any ideas.

**Reply by laymonage** [reply]
Thanks, glad you got it up and running so quickly!

The page reload is necessary because it isn't actually a page reload: you're being redirected to GitHub, which then redirects you to the original page. You're already signed in and approved giscus on your account, so you didn't see anything. For new users, they'll get a dialog on GitHub asking for permission.

I think your project is cool as well. I like the clean and straightforward look. Is it customizable? Maybe some people would like to use different fonts/sizes etc. I have yet to use Ruby on my projects, but I'll definitely consider using your project when I do. Also, I noticed that the "Documentation Formatting" link on the homepage is broken: https://socketry.github.io/utopia-project

As for my project, for now I'm still sticking to GitHub's aesthetics. It's still in active development so feel free to add suggestions at https://github.com/laymonage/giscus.
